app/src/main/java/ch/epfl/sdp/peakar/user/profile/NewProfileActivity.java

```diff
@@ -28,20 +40,59 @@
  * TODO Rename to remove new part.
  */
 public class NewProfileActivity extends AppCompatActivity {
+    public final static String AUTH_INTENT = "isAuth";
+    public final static String OTHER_INTENT = "otherId";
+
+    private boolean isAuthProfile;
+
+    private OtherAccount otherAccount = null;
+
 
     private View selectedCollected = null;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_new_profile);
 
-        StatusBarHandler.StatusBarTransparent(this);
+        // Get the starting intent
+        Intent startingIntent = getIntent();
+        isAuthProfile = startingIntent.getBooleanExtra(AUTH_INTENT, false);
+
+        if(!isAuthProfile) {
+            String otherId = startingIntent.getStringExtra(OTHER_INTENT);
+            new Thread(() -> {
+                otherAccount = OtherAccount.getInstance(otherId);
+                runOnUiThread(() -> {
+                    setContentView(R.layout.activity_new_profile);
+
+                    StatusBarHandler.StatusBarTransparent(this);
+
+                    hideUI(false, true);
+                    setupProfile(otherAccount.getUsername(), (int)otherAccount.getScore());
 
-        hideUI(false, false);
-        setupProfile("Alexander", 4978);
+                    fillListView();
+                });
+
+            }).start();
+        } else {
+            setContentView(R.layout.activity_new_profile);
+
+            StatusBarHandler.StatusBarTransparent(this);
+
+            hideUI(true, false);
+            setupProfile(AuthService.getInstance().getAuthAccount().getUsername(), (int)AuthService.getInstance().getAuthAccount().getScore());
+
+
+            // If the user is not registered, force a username change
+            if(AuthService.getInstance().getAuthAccount().getUsername().equals(Account.USERNAME_BEFORE_REGISTRATION)) {
+                changeUsernameButton(null);
+                // Set text view
+                ((TextView)findViewById(R.id.profile_empty_text)).setText(R.string.empty_collection_not_registered);
+            } else {
+                fillListView();
+            }
+        }
 
```

Comment:
Add a listener for keyboard changes to handle the case when the keyboard is open and the user presses the back button.
```suggestion
        findViewById(R.id.profile_collection).setOnApplyWindowInsetsListener((v, insets) -> {
            if(!isKeyboardOpened()){
                setUiUsernameChanged();
            }
            return insets;
        });
```